test/suites/gpu/suite_test.go

```diff
@@ -51,21 +51,32 @@ var _ = AfterEach(func() { env.Cleanup() })
 var _ = AfterEach(func() { env.AfterEach() })
 
 var _ = Describe("GPU", func() {
-	// Table test for gpu passing in different node classes
+	// Table test for GPU provisioning
 	DescribeTable("should provision one GPU node and one GPU Pod",
 		func(nodeClass *v1alpha2.AKSNodeClass) {
+			// Enable NodeRepair feature gate if running in-cluster
+			if env.InClusterController {
+				// One scenario we want to test, is that our node-auto-repair isn't too aggressive.
+				// If a gpu vm joins the cluster at 3 minutes, and takes 10 minutes to get ready, we don't want
+				// to GC the vm for node auto repair.
+				// By enabling the controller, the below assertions that the gpu workload eventually gets scheduled
+				// and runs means we didnt remove the not-ready vm
+				env.ExpectSettingsOverridden(corev1.EnvVar{Name: "FEATURE_GATES", Value: "NodeRepair=True"})
```

Comment:
we should pick to either say corev1, or v1, I would opt for corev1 since there are so so many v1 packages.